core/kernel/src/c_interface.rs

```diff
@@ -0,0 +1,129 @@
+use crate::{
+    feature_manifest::{FeatureManifest, FeatureManifestParsingError},
+    kernel::Kernel,
+    kernel::KERNEL,
+};
+use serde::Serialize;
+use std::{
+    ffi::{CStr, CString},
+    os::raw::c_char,
+    str::Utf8Error,
+};
+
+/// # Safety
+/// This function can be unsafe if the language_code pointer is null or the string is in wrong format.
+///
+/// Bootstrap the kernel with the given configuration:
+/// - language_code: User's locale language code
+/// Returns the result JSON of either success or failure to bootstrap the kernel
+#[no_mangle]
+pub unsafe extern "C" fn kernel_bootstrap(language_code: *const c_char) -> *const c_char {
+    unsafe fn bootstrap(language_code: *const c_char) -> Result<(), KernelError> {
+        let language_code = cstring_to_str(&language_code).map(String::from)?;
+        Kernel::bootstrap(language_code).map_err(KernelError::FailedToBootstrapKernel)
+    }
+
+    to_result_str(bootstrap(language_code))
+}
+
+/// # Safety
+/// This function can be unsafe if the json pointer is null or the string is in wrong format.
+///
+/// Parse the feature manifest from a given JSON.
+/// Returns the JSON with the parsed feature manifest by taking into account the user's locale language code.
+///
+/// This is unfortunate at the moment that this API gets a JSON and responds back with a JSON. Ideally the kernel
+/// would be able to read all the features from the disk on its own and return back a JSON with all of the features.
+/// It was choosen to return back to platform code a JSON, as composing complex C model is not nice. For example
```

Comment:
But this is a problem. Either
1. We consume the data structure locally
2. We use some other bridge mechanism
3. We find a way to represent complex data structures in C, maybe making bridge code a bit more complex, with its own libraries.

Reply:
Added FlatBuffers.